### PR TITLE
Add metrics for verticles

### LINKED ISSUE #36: Add metrics for verticles
- Total number of verticles deployed
- Number of instances per verticle deployed

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #36

### CODE REVIEW

**Comment by tsegismont** [comment]
@cescoffier can you please review? Thanks

File: src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static java.util.stream.Collectors.*;
+
+/**
+ * @author Thomas Segismont
+ */
+public class VerticleMetricsSupplier implements MetricSupplier {
+  private final ConcurrentMap<String, Integer> verticleInstances = new ConcurrentHashMap<>(0);
```
Comment (vietj) [inline src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java]: why initial capacity is `0` ?

File: src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      .map(entry -> new GaugePoint(baseName + entry.getKey(), timestamp, entry.getValue().doubleValue()))
+      .collect(toList());
+  }
+
+  public void verticleDeployed(Verticle verticle) {
+    verticleInstances.compute(nameOf(verticle), (name, val) -> val == null ? 1 : val + 1);
+  }
+
+  private String nameOf(Verticle verticle) {
+    return verticle.getClass().getName();
```
Comment (vietj) [inline src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java]: why not include `baseName` in the name of a verticle which avoids to append it when creating `GaugePoint` instances ?

**Review by vietj — COMMENTED** [review]
I've done a couple of comments, besides it looks good

File: src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java (hunk 2 above)
Comment (cescoffier) [reply src/main/java/io/vertx/ext/hawkular/impl/VerticleMetricsSupplier.java]: +1

**Comment by cescoffier** [comment]
I share Julien's comments. Looks good to me and a great addition.

**Comment by tsegismont** [comment]
By habit to avoid creating it with the default size

2017-06-07 20:46 GMT+02:00 Julien Viet <notifications@github.com>:

**Comment by tsegismont** [comment]
Will do

2017-06-07 20:49 GMT+02:00 Julien Viet <notifications@github.com>:

**Comment by tsegismont** [comment]
@vietj @cescoffier PR updated. Thank you!

**Comment by tsegismont** [comment]
Approved off-list by @vietj

**Pull request merged**